### PR TITLE
Sanitize url before fetching favicon and display letter avatar if it can't be retrieved

### LINKED ISSUE #919: Sanitize url before fetching favicon and display letter avatar if it can't be retrieved
# Problem

When I paste a link from Google Omnibox ex:(http://google.com/) it inserts a "/" at the end of the URL and "https://" at the beginning of the URL. **This prevents the favicon to get fetched**

Image

Apple favicon doesn't get fetched if no "www" is added in front of the domain name, instead, it gives a "little temple"

Image

# Solution

## Api call

We should try fetching the favicon in cascade with different URLs formats, in that order:

1) "example.com" by removing any "https://" "www" and the "/" at the end of the sentence.
2) "www.example.com"
3) "https://www.example.com"

Image

## Display

In the table, we should display the URL in "example.com" format (no "https", no "www") to make it easy to read.

## In case of no results

if no results can be fetched, we should insert the following placeholders:

Image
Image

Figma components:
https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=3625-52525&mode=design&t=kDoI67TzhyFkHyn1-4
https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=808-51105&mode=design&t=kDoI67TzhyFkHyn1-4

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #919 

Loom: 
https://www.loom.com/share/529f6fe129cf4868880e90ca4f54a099?sid=1f9baa3d-973a-4c8a-8a99-ce96c3495069

Happy hacking! 🎉

### CODE REVIEW

File: front/src/utils/index.ts [hunk 1]
```diff
@@ -23,7 +23,13 @@ export function getLogoUrlFromDomainName(
...
   if (!sanitizedDomain) return;
 
-  for (const prefix of ['', 'www.', 'https://']) {
+  const prioritizeWWW = ['apple.com'].includes(sanitizedDomain);
+
+  const prefixes = prioritizeWWW
+    ? ['www.', '', 'https://']
+    : ['', 'www.', 'https://'];
+
+  for (const prefix of prefixes) {
```
Comment (gitstart-twenty) [inline front/src/utils/index.ts:32]: It appears that when searching for the apple favicon using a URL without the `www` prefix, it returns the `temple` favicon instead. To address this issue, we decided to switch it up and begin the search with the `www` prefix. However, we recognize that this solution is not scalable as we cannot maintain different priorities for various websites.

File: front/src/utils/__tests__/utils.test.ts [hunk 2]
```diff
@@ -1,21 +1,56 @@
...
+    expect(sanitizeURL(undefined)).toBe('');
+  });
+});
 
 describe('getLogoUrlFromDomainName', () => {
-  it(`should generate logo url if undefined `, () => {
-    expect(getLogoUrlFromDomainName(undefined)).toBe(
-      'https://api.faviconkit.com/undefined/144',
+  test('should return the correct logo URL for a given domain', () => {
+    const imgMock = {
```
Comment (charlesBochet) [inline front/src/utils/__tests__/utils.test.ts:19]: i don't think this is s actually useful. Let's just test getLogoUrlFromDomainName

File: front/src/utils/index.ts (hunk 1 above)
Comment (charlesBochet) [reply front/src/utils/index.ts:32]: Ok, we are putting in place our own favicon server to fix the issue at the root cause :)

Comment (charlesBochet) [reply front/src/utils/index.ts:32]: favicon.twenty.com/microsoft.com

File: front/src/utils/index.ts [hunk 3]
```diff
@@ -10,6 +10,36 @@ export function formatToHumanReadableDate(date: Date | string) {
...
+  if (!sanitizedDomain) return;
+
+  const prioritizeWWW = ['apple.com'].includes(sanitizedDomain);
+
+  const prefixes = prioritizeWWW
+    ? ['www.', '', 'https://']
+    : ['', 'www.', 'https://'];
+
+  for (const prefix of prefixes) {
+    const img = document.createElement('img');
```
Comment (charlesBochet) [inline front/src/utils/index.ts:33]: It seems weird to create a fake image here. Let's use fetch api instead if possible

File: front/src/utils/__tests__/utils.test.ts (hunk 2 above)
Comment (charlesBochet) [reply front/src/utils/__tests__/utils.test.ts:19]: Ok, understood, let's use fetch instead (see comment below)

File: front/src/utils/index.ts [hunk 4]
```diff
@@ -10,6 +10,36 @@ export function formatToHumanReadableDate(date: Date | string) {
...
+}
+
+export function getLogoUrlFromDomainName(
+  domainName?: string,
+): string | undefined {
+  const sanitizedDomain = sanitizeURL(domainName);
+
+  if (!sanitizedDomain) return;
+
+  const prioritizeWWW = ['apple.com'].includes(sanitizedDomain);
```
Comment (charlesBochet) [inline front/src/utils/index.ts:26]: let's only do the sanitzing, we don't need to query the www domain as we are going to use favicon.twenty.com that will always return an icon for www and root domains

**Comment by charlesBochet** [comment]
@gitstart-twenty Sorry about the strategy change on this one (we are going to implement favicon.twenty.com here in mid term). I think we can still merge this one, I have a few comments on it

File: front/src/modules/users/components/Avatar.tsx [hunk 5]
```diff
@@ -95,7 +110,8 @@ export function Avatar({
       type={type}
       colorId={colorId}
     >
-      {noAvatarUrl && placeholder[0]?.toLocaleUpperCase()}
+      {(noAvatarUrl || isInvalidAvatarUrl) &&
+        placeholder[0]?.toLocaleUpperCase()}
```
Comment (gitstart-twenty) [inline front/src/modules/users/components/Avatar.tsx:114]: Hey @charlesBochet,
How about we let the avatar determine whether to display a letter or an image? I believe this is a lot simpler.

Comment (charlesBochet) [reply front/src/modules/users/components/Avatar.tsx:114]: Yes, that makes sense

**Review by charlesBochet — APPROVED** [review]
Thank you :)

**Pull request merged**